lib/snapshot.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+const path = require('path');
+const fs = require('fs');
+const isEqual = require('lodash.isequal');
+const mkdirp = require('mkdirp');
+const globals = require('./globals');
+
+class Snapshot {
+	constructor(testPath, options) {
+		if (!testPath) {
+			throw new TypeError('Test file path is required');
+		}
+
+		this.dirPath = path.join(path.dirname(testPath), '__snapshots__');
+		this.filePath = path.join(this.dirPath, path.basename(testPath) + '.snap');
+		this.tests = {};
+		this.options = options || {};
+
+		if (fs.existsSync(this.filePath)) {
+			this.tests = JSON.parse(fs.readFileSync(this.filePath));
+		}
+	}
+
+	save() {
+		mkdirp.sync(this.dirPath);
```

Comment:
We should add something like:

```javascript
let hasTests = false;
for(var key in obj) {
  if(obj.hasOwnProperty(key))
    hasTests = true;
    break;
  }
}
if (!hasTests) return;
```

at the beginning of this function. That way we don't end up making a bunch of empty files/divs.

We also need to ask ourselves what we want to do if someone deletes a test. Should we delete the snapshots too? Or leave them there?